### DISCUSSION TITLE
Can "UVtoolsCmd convert" read FILEFORMAT_? in sl1 files?

### DISCUSSION THREAD

**Discussion by msspad** [sn4k3/UVtools#522] [Q&A]
When opening sl1 files with UVtools, it can read FILEFORMAT_CTB in the notes of a sl1 file and convert accordingly. Is there a way to tell 
```UVtoolsCmd convert myfile.sl1``` to read FILEFORMAT_x in the same way? When no target-type is provided, UVtoolsCmd throws an error.

Certainly not the highest priority but could you copy over the sl1.MaterialName property in some way to the ctb file format when converting with UVtoolsCmd ? When sl1 files just got converted in UVtools gui, scripts can read sl1.MaterialName and append it to ctb.MachineName. Any other solution would be fine too.

**Comment by sn4k3** [comment]
The auto conversion feature is exclusive and controlled by the UI to help user with PrusaSlicer, is not part of the core main `Convert` function. 
The convert command both on Core and Cmd always require a target type. I will not change the default behaviour of cmd-convert but i can add a flag like `--auto-convert` if that's important to have, you tell me. Still limited only to SL1 and VDT files.

> could you copy over the sl1.MaterialName property in some way to the ctb file format

All properties are [transferred](https://github.com/sn4k3/UVtools/blob/315ce3bc0dfc7a950138473f66ac83e0fa119f6c/UVtools.Core/FileFormats/FileFormat.cs#L4732), however when you close the file and the target format lacks a field to store that information, it is lost. Since there's no field to store MaterialName on ctb there's no way to keep it after close the file. However you can create a script do dump the material name with `print-properties` command and rename file in same script or pass directly to the `convert` command. eg: `UVtoolsCmd.exe print-properties .\UVtools_demo_file.sl1 --partial-mode -bn MaterialName`.
Still you will need a PS script if on Windows or bach if on Linux.
Output:

```bash
PS C:\Program Files\UVtools> .\UVtoolsCmd.exe print-properties .\UVtools_demo_file.sl1 --partial-mode -bn MaterialName
Opening file UVtools_demo_file.sl1: Done in 0.11s
Listing properties:
----------------------
MaterialName: BlueCast EcoGray @0.1
----------------------
Total properties: 1
```

Regex: https://regex101.com/r/ridCH7/1

**Reply by msspad** [reply]
yes, ```--auto-convert``` would be useful as it enables the same workflow as with the GUI.

```UVtoolsCmd.exe print-properties --partial-mode``` is an option, thanks.

**Reply by sn4k3** [reply]
It's done:

```batch
PS net6.0> .\UVtoolsCmd.exe convert .\test.ctb -?
Description:
  Convert input file into a output file format by a known type or extension

Usage:
  UVtoolsCmd convert [<input-file> <target-type/ext> [<output-file>]] [options]

Arguments:
  <input-file>       Input file to open and read
  <target-type/ext>  Target format type or extension. Use 'auto' for SL1 files with specified FILEFORMAT_xxx
  <output-file>      Output file to save []

Options:
  -v, --version <version>  Sets the file format version
  --no-overwrite           If the output file exists do not overwrite
  -?, -h, --help           Show help and usage information


PS net6.0> .\UVtoolsCmd.exe convert .\test.ctb auto
Opening file test.ctb: Done in 0.14s
Error: The file 'test.ctb' is not a valid candidate for auto conversion. Please specify the target format instead.

PS net6.0> .\UVtoolsCmd.exe convert .\test.sl1 auto
Opening file test.sl1: Done in 0.14s
Opening file test.sl1: Done in 0.06s
Converting to test.cxdlp: Done in 0.22s
```

**Reply by msspad** [reply]
looks perfect, thanks!

**Reply by sn4k3** [reply]
Check new version

**Reply by msspad** [reply]
works like a charm, thanks!

I don't really care but when opening an encrypted.ctb, the explainer text is hidden and only reachable with scrolling
![image](https://user-images.githubusercontent.com/6191764/181903097-da72f206-4dfd-4f30-a997-e9091fb9019a.png)

**Reply by sn4k3** [reply]
That's strange, it also show for me like that.
I'm always fighting this message boxes... So annoying.
